x-pack/legacy/plugins/epm/server/packages/assets.ts

```diff
@@ -40,26 +46,23 @@ export function getAssets(
   return assets;
 }
 
-export function getAssetsData(
+export async function getAssetsData(
   packageInfo: RegistryPackage,
   filter = (path: string): boolean => true,
   dataSet: string = ''
-): Registry.ArchiveEntry[] {
+): Promise<Registry.ArchiveEntry[]> {
+  // TODO: Needs to be called to fill the cache but should not be required
```

Comment:
We could probably remove this comment here as it was odd having this extra request in the template code but here in assets it makes more sense, especially as it prechecks the cache.